File: shoring/compat/src/test/java/com/panopset/compat/test/ZombieTest.java
package com.panopset.compat.test;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.panopset.compat.Zombie;

public class ZombieTest {
  String foo = "bar";

  @Test
  void test() {
    Zombie zombie = new Zombie();
    Assertions.assertTrue(zombie.isActive());
    Assertions.assertEquals("bar", foo);
    zombie.addStopAction(new Runnable() {
      @Override
      public void run() {
        ZombieTest.this.foo = "bat";
      }
    });
    Assertions.assertEquals("bar", foo);
    Assertions.assertTrue(zombie.isActive());
    Assertions.assertEquals("bar", foo);
    Assertions.assertTrue(zombie.isActive());
    zombie.stop();
    Assertions.assertFalse(zombie.isActive());
    Assertions.assertEquals("bat", foo);
    zombie.stop();
    Assertions.assertFalse(zombie.isActive());
    Assertions.assertEquals("bat", foo);
    zombie.resume();
    Assertions.assertTrue(zombie.isActive());
  }
}

File: code/shoring/compat/src/test/kotlin/com/panopset/compat/test/ZombieTest.kt
package com.panopset.compat.test

import com.panopset.compat.Zombie
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class ZombieTest {
    var foo = "bar"
    @Test
    fun test() {
        Assertions.assertTrue(Zombie.isActive)
        Assertions.assertEquals("bar", foo)
        Zombie.addStopAction { foo = "bat" }
        Assertions.assertEquals("bar", foo)
        Assertions.assertTrue(Zombie.isActive)
        Assertions.assertEquals("bar", foo)
        Assertions.assertTrue(Zombie.isActive)
        Zombie.stop()
        Assertions.assertFalse(Zombie.isActive)
        Assertions.assertEquals("bat", foo)
        Zombie.stop()
        Assertions.assertFalse(Zombie.isActive)
        Assertions.assertEquals("bat", foo)
        Zombie.resume()
        Assertions.assertTrue(Zombie.isActive)
    }
}
